tests/rptest/services/franz_go_verifiable_services.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2022 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import os
+import threading
+from ducktape.services.background_thread import BackgroundThreadService
+
+# The franz-go root directory
+TESTS_DIR = os.path.join("/opt", "si-verifier")
+
+from enum import Enum
+
+
+class ServiceStatus(Enum):
+    SETUP = 1
+    RUNNING = 2
+    FINISH = 3
+
+
+class FranzGoVerifiableService(BackgroundThreadService):
+    """
+    FranzGoVerifiableService is si-verifier service.
+    To validate produced record user should run consumer and producer in one node.
+    Use ctx.cluster.alloc(ClusterSpec.simple_linux(1)) to allocate node and pass it to constructor
+    """
+    def __init__(self, context, redpanda, topic, msg_size, custom_node):
+        self.use_custom_node = custom_node is not None
+
+        # We should pass num_nodes to allocate for our service in BackgroundThreadService,
+        # but if user allocate node by themself, BackgroundThreadService should not allocate any nodes
+        nodes_for_allocate = 1
+        if self.use_custom_node:
+            nodes_for_allocate = 0
+
+        super(FranzGoVerifiableService,
+              self).__init__(context, num_nodes=nodes_for_allocate)
+
+        # Should check that BackgroundThreadService did not allocate anything
+        # and store allocated nodes by user to self.nodes
+        if self.use_custom_node:
+            assert not self.nodes
+            self.nodes = custom_node
```

Comment:
just for discussion:

should we think about creating a `ColocatedBackgroundService` helper class? it seems like there a fair amount of complexity in this test harness unrelated to running si-verifier and specifically about dealing resource management. no need to do it in this PR, but we should always be looking for opportunities like this where other devs are probably going to encounter a similar need.

For example:

```python
        self._node_for_franz_go = ctx.cluster.alloc(
              ClusterSpec.simple_linux(1))
 
          self._producer = FranzGoVerifiableProducer(
              ctx, self.redpanda, self.topic, FranzGoVerifiableTest.MSG_SIZE,
              100000, self._node_for_franz_go)
          self._seq_consumer = FranzGoVerifiableSeqConsumer(
              ctx, self.redpanda, self.topic, FranzGoVerifiableTest.MSG_SIZE,
              self._node_for_franz_go)
          self._rand_consumer = FranzGoVerifiableRandomConsumer(
              ctx, self.redpanda, self.topic, FranzGoVerifiableTest.MSG_SIZE,
              1000, 20, self._node_for_franz_go)
```

Looks like an opportunity to create a meta service runner that can take care of allocation...

Reply:
Looks so useful it this case